transcom/mymove
# INTEGRATION-B -19167 - Feature flags to hide/show specific shipment type - Customer UI

## [B-19167](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A923722)

## Summary

**INTEGRATION**

This PR introduces feature flags to show/hide specific shipment types on the customer application side. It is in preparation for the upcoming production release of Milmove in early April. Dan Shuester wants to only enable HHG for the production release and disable PPM, NTS, NTSR. Note: this is only achieved via UI and NOT on the server side because CreateMtoShipments API is still opened to creating PPM, NTS, NTSR from the Prime/Office side. Dan Shuester says this is acceptable.  Specific shipment types will be shown/hidden on the "Add Shipment" view per configuration.

[Feature Flag](https://transcom.github.io/mymove-docs/docs/tools/feature-flags/how-to-use-feature-flags/#introduction)

.envrc
export FEATURE_FLAG_PPM=true
export FEATURE_FLAG_NTS=true
export FEATURE_FLAG_NTSR=true

Note: no configuration for HHG

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

By default  all shipment types are enabled via  .envrc:

export FEATURE_FLAG_PPM=true
export FEATURE_FLAG_NTS=true
export FEATURE_FLAG_NTSR=true

Note: no configuration for HHG

1. Update .envrc to hide/show specific shipment type 
2. Start app and login into customer 
3. Start adding shipments ..etc...
4. Verify INFO modal popups are consistent with what are enabled/disabled. ie...It should not reference PPM if PPM is disabled ..etc...

If everything is enabled it should should display the following:
![Screenshot 2024-03-18 at 4 24 21 PM](https://github.com/transcom/mymove/assets/148127004/8fc5c900-20c6-4df8-91d7-839b5bc6c50b)
![Screenshot 2024-03-18 at 6 01 34 PM](https://github.com/transcom/mymove/assets/148127004/e5b9e560-f027-4e69-a559-db70c3f42075)
![Screenshot 2024-03-18 at 9 28 32 PM](https://github.com/transcom/mymove/assets/148127004/c45fedd9-7962-4f49-8217-ae49f648ca2a)
<img width="918" alt="Screenshot 2024-03-19 at 8 44 50 AM" src="https://github.com/transcom/mymove/assets/148127004/add1761e-5b83-4152-a9d2-24c8c73ac653">

### Frontend


### Backend
